File: GATEST12/app/src/main/java/com/example/gatest12/AnalyticsApplication.java
package com.example.gatest12;

import android.app.Application;

import com.google.android.gms.analytics.GoogleAnalytics;
import com.google.android.gms.analytics.Tracker;

public class AnalyticsApplication extends Application {

    private static Tracker sTracker;
    private static GoogleAnalytics sAnalytics;
    @Override
    public void onCreate() {
        super.onCreate();
        sAnalytics = GoogleAnalytics.getInstance(this);
    }

    synchronized public Tracker getDefaultTracker(){
        if(sTracker == null){
            sTracker = sAnalytics.newTracker(R.xml.global_tracker);
            sTracker.enableAdvertisingIdCollection(true);
            sTracker.enableExceptionReporting(true);
            sTracker.setAnonymizeIp(true);

        }
        return sTracker;
    }
}

File: GP_Kotlin-master/GP_Kotlin-master/app/src/main/java/app/suro/myapplication/AnalyticsApplication.kt
package app.suro.myapplication

import android.app.Application
import com.google.android.gms.analytics.GoogleAnalytics
import com.google.android.gms.analytics.Tracker

class AnalyticsApplication : Application() {
    private var sAnalytics: GoogleAnalytics? = null
    private var sTracker : Tracker? = null

    override fun onCreate() {
        super.onCreate()
        sAnalytics = GoogleAnalytics.getInstance(this)
    }

    @Synchronized
    fun getDefaultTracker(): Tracker? {
        if(sTracker == null) {
            sAnalytics?.let {
                sTracker = it.newTracker(R.xml.global_tracker)
                sTracker?.enableAdvertisingIdCollection(true)
                sTracker?.enableExceptionReporting(true)
                sTracker?.setAnonymizeIp(true)
            }
        }
        return sTracker
    }
}